src/pytorch_lightning/serve/servable_module_validator.py

```diff
@@ -0,0 +1,175 @@
+import logging
+import time
+from multiprocessing import Process
+from typing import Any, Dict, Optional
+
+import requests
+import torch
+from typing_extensions import Literal
+
+import pytorch_lightning as pl
+from pytorch_lightning.callbacks import Callback
+from pytorch_lightning.serve.servable_module import ServableModule
+from pytorch_lightning.strategies import DDPFullyShardedNativeStrategy, DDPFullyShardedStrategy, DeepSpeedStrategy
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from pytorch_lightning.utilities.imports import _RequirementAvailable
+from pytorch_lightning.utilities.model_helpers import is_overridden
+from pytorch_lightning.utilities.rank_zero import rank_zero_only
+
+_NOT_SUPPORTED_STRATEGIES = (
+    DeepSpeedStrategy,
+    DDPFullyShardedNativeStrategy,
+    DDPFullyShardedStrategy,
+)
+
+_logger = logging.getLogger(__name__)
+
+
+class ServableModuleValidator(Callback):
+    """The ServableModuleValidator validates to validate a model correctly implement the ServableModule API.
+
+    .. warning::
+
+        This is currently an experimental feature and API changes are to be expected.
+
+    Arguments:
+        optimization: The format in which the model should be tested while being served.
+        server: The library used to evaluate the model serving. Default is FastAPI.
+        host: The host address associated the server.
+        port: The port associated the server.
+    """
+
+    def __init__(
+        self,
+        optimization: Optional[Literal["trace", "script", "onnx", "tensorrt"]] = None,
+        server: Literal["fastapi", "ml_server", "torchserve", "sagemaker"] = "fastapi",
+        host: str = "127.0.0.1",
+        port: int = 8080,
+        timeout: int = 10,
+        exit_on_failure: bool = True,
+    ):
+        super().__init__()
+        fastapi_installed = _RequirementAvailable("fastapi")
```

Comment:
Probably outside the scope of this PR, but would it make sense to save the full set of dependencies for a model after training so that we can check whether these are present on the serving environment? I think that this is something that the user could declare and then Lightning could save after training.

Reply:
This is 100 % a great idea. I think more metadata should be stored, like requirements, payload, etc...

Reply:
Probably integrates quite nice with https://pytorch.org/docs/stable/package.html